src/shell/shell_cmds.cpp

```diff
@@ -740,6 +741,74 @@ void DOS_Shell::CMD_DIR(char * args) {
 	dos.dta(save_dta);
 }
 
+void DOS_Shell::CMD_LS(char *args)
+{
+	HELP("LS");
+
+	RealPt save_dta = dos.dta();
+	dos.dta(dos.tables.tempdta);
+	DOS_DTA dta(dos.dta());
+
+	const std::string pattern = to_search_pattern(args);
+	bool ret = DOS_FindFirst(pattern.c_str(), 0xffff & ~DOS_ATTR_VOLUME);
+	if (!ret) {
+		WriteOut(MSG_Get("SHELL_CMD_LS_PATH_ERR"), trim(args));
+		dos.dta(save_dta);
+		return;
+	}
+
+	std::vector<DtaResult> results;
+	// reserve space for as many as we can fit into a single memory page
+	// nothing more to it; make it larger if necessary
+	results.reserve(4096 / sizeof(DtaResult));
```

Comment:
Can we use either:
- **include/mem.h**: `#define MEM_PAGESIZE 4096`
- **include/paging.h**: `#define MEM_PAGE_SIZE  (4096)`